pkg/resource/settings/delete_test.go

```diff
@@ -0,0 +1,482 @@
+			Type:           "builtin:some-other-schema",
+			OriginObjectId: "some-other-id",
+		}
+
+		c := deleteStubClient{
+			list: func(schema string, options dtclient.ListSettingsOptions) ([]dtclient.DownloadSettingsObject, error) {
+				var givenObject = dtclient.DownloadSettingsObject{
+					SchemaId: given.Type,
+					ObjectId: given.OriginObjectId,
+				}
+
+				var otherObject = dtclient.DownloadSettingsObject{
+					SchemaId: other.Type,
+					ObjectId: other.OriginObjectId,
+				}
+
+				if options.Filter(givenObject) {
+					return []dtclient.DownloadSettingsObject{givenObject}, nil
+				} else if options.Filter(otherObject) {
+					return []dtclient.DownloadSettingsObject{otherObject}, nil
+				}
+
+				return []dtclient.DownloadSettingsObject{}, nil
+			},
+			delete: func(id string) error {
+				if id == given.OriginObjectId {
+					return nil
+				}
+				return errors.New("some unpredictable error")
+			},
+		}
+
+		err := settings.NewDeleter(&c).Delete(t.Context(), []pointer.DeletePointer{given, other, given}) // the pointer in the middle is to cause error behavior
+		assert.ErrorContains(t, err, "failed to delete 1 settings objects(s)")
+		assert.True(t, c.deleteCalled)
+	})
+}
+
+func TestDeleteAll(t *testing.T) {
+	t.Run("simple case", func(t *testing.T) {
+		c := deleteStubClient{
+			listSchemas: func() (dtclient.SchemaList, error) {
+				return []dtclient.SchemaItem{
+					{SchemaId: "builtin:some-schema"},
+					{SchemaId: "builtin:some-other-schema"},
+				}, nil
+			},
+			list: func(schema string, options dtclient.ListSettingsOptions) ([]dtclient.DownloadSettingsObject, error) {
+				if schema == "builtin:some-schema" {
+					return []dtclient.DownloadSettingsObject{
+						{
+							SchemaId: "builtin:some-schema",
+							ObjectId: "some-object-1",
+						},
+						{
+							SchemaId: "builtin:some-schema",
+							ObjectId: "some-object-2",
+						},
+					}, nil
+				} else if schema == "builtin:some-other-schema" {
+					return []dtclient.DownloadSettingsObject{
+						{
+							SchemaId: "builtin:some-other-schema",
+							ObjectId: "some-object-3",
+						},
+					}, nil
+				}
+
+				return []dtclient.DownloadSettingsObject{}, nil
+			},
+			delete: func(uid string) error {
+				assert.Contains(t, []string{"some-object-1", "some-object-2", "some-object-3"}, uid)
+				return nil
+			},
+		}
+
+		err := settings.NewDeleter(&c).DeleteAll(t.Context())
+		assert.NoError(t, err)
+		assert.True(t, c.deleteCalled)
+	})
+
+	t.Run("deletion continues even if error occurs during delete", func(t *testing.T) {
+		c := deleteStubClient{
+			listSchemas: func() (dtclient.SchemaList, error) {
+				return []dtclient.SchemaItem{
+					{SchemaId: "builtin:some-schema"},
+					{SchemaId: "builtin:some-other-schema"},
+				}, nil
+			},
+			list: func(schema string, options dtclient.ListSettingsOptions) ([]dtclient.DownloadSettingsObject, error) {
+				responses := map[string][]dtclient.DownloadSettingsObject{
```

Comment:
We could move this outside this test run, because the one above uses the same objects, and therefore it can be reused, and also because the responses don't need to be re-created on every list call